lib/throttler.js

```diff
@@ -10,86 +11,107 @@ var defaults      = require('../defaults'),
+  return throttleMiddleware
+
+  function errHandler (res, next, error) {
+    // DO WE WANT TO BREAK THE APP AND CAUSE A ISSUE OR DO WE WANT TO SILENTLY INTO THE NIGHT AND JUST LOG OUR ERRORS
+    // SEND A EMAIL POTENTIALLY IF SOMEHTING IS THAT BROKEN OR IF WE SENSE A PEN TEST OF 10000+ REQUESTS
+    if (error) {
+      res.statusCode = 500
+      next(error)
+      return true
     }
-    if (typeof limited != 'function') {
-        limited = false;
+  }
+
+  function respondWithThrottle (request, response, next, throttle) {
+    var timeUntilReset = (defaults.rateLimit.ttl * 1000) - (new Date().getTime() - throttle.createdAt.getTime())
+    var remaining = Math.max(0, (defaults.rateLimit.max - throttle.hits))
+
+    response.set({
+      'X-Rate-Limit-Limit': defaults.rateLimit.max,
+      'X-Rate-Limit-Remaining': remaining,
+      'X-Rate-Limit-Reset': timeUntilReset
+    })
+
+    if (throttle.hits < defaults.rateLimit.max) {
+      return next()
     }
 
-    config                  = config                  || defaults;
-    config.rateLimit        = config.rateLimit        || defaults.rateLimit;
-    config.rateLimit.ttl    = config.rateLimit.ttl    || defaults.rateLimit.ttl;
-    config.rateLimit.max    = config.rateLimit.max    || defaults.rateLimit.max;
-    config.response         = config.response         || defaults.response;
-    config.response.code    = config.response.code    || defaults.response.code;
-    config.response.message = config.response.message || defaults.response.message;
+    response.statusCode = defaults.response.code
+    if (limited) {
+      return limited(request, response, throttle.hits, timeUntilReset)
+    }
 
-    Throttle = throttleModel(config);
-    return throttle_middleware;
+    return response.send(defaults.response.message)
+  }
+  /**
+   * Check for request limit on the requesting IP
+   *
+   * @access public
+   * @param {object} request Express-style request
+   * @param {object} response Express-style response
+   * @param {function} next Express-style next callback
+   */
 
-/**
- * Check for request limit on the requesting IP
- *
- * @access public
- * @param {object} request Express-style request
- * @param {object} response Express-style response
- * @param {function} next Express-style next callback
- */
-function throttle_middleware(request, response, next) {
-    'use strict';
+  function throttleMiddleware (request, response, next) {
     var ip = request.headers['x-forwarded-for'] ||
-        request.connection.remoteAddress ||
-        request.socket.remoteAddress ||
-        request.connection.socket.remoteAddress;
+      request.connection.remoteAddress ||
+      request.socket.remoteAddress ||
+      request.connection.socket.remoteAddress || ''
+
+    ip = ip.split(',')[0]
+    if (ip.substr(0, 7) === '::ffff:' || ip === '::1') {
+      // DEFUALT TO LOCALHOST SO THAT IF WE DO GET A INVALID IP WE CAN RELY ON THE WE CAN STILL LIMIT THE ATTACK OR PERSON
```

Comment:
see https://github.com/andjosh/mongo-throttle/commit/7b52d4b07ea45616a84da721dd605b999e9fb60e (and 60ff0ee72fe1f242763d67b64e7f7561e5bdfb1d)